### PR TITLE
feat(fs): support FileInfo.dev on Windows

### PR DESCRIPTION
**Pull request by bartlomieju**
This commit adds support for retrieving `dev` information
when stating files on Windows.

Additionally `Deno.FileInfo` interfaces was changed to always
return 0 for fields that we don't retrieve information for on Windows.

Closes https://github.com/denoland/deno/issues/18053

### CODE REVIEW

File: ext/fs/30_fs.js [hunk 1]
```diff
@@ -211,31 +211,16 @@ async function rename(oldpath, newpath) {
...
 //  high u32 | low u32
-//
-// 4. ?u64 converts a zero u64 value to JS null on Windows.
 function createByteStruct(types) {
   // types can be "date", "bool" or "u64".
-  // `?` prefix means optional on windows.
   let offset = 0;
-  let str =
-    'const unix = Deno.build.os === "darwin" || Deno.build.os === "linux"; return {';
+  let str = "return {";
```
Comment (aapoalas) [inline ext/fs/30_fs.js:217]: I wonder if this whole eval thingamajig is even necessary now?

File: ext/fs/lib.rs [hunk 2]
```diff
@@ -1226,13 +1226,169 @@ fn get_stat(metadata: std::fs::Metadata) -> FsStat {
...
+      path.as_ptr(),
+      0,
+      FILE_SHARE_READ | FILE_SHARE_DELETE | FILE_SHARE_WRITE,
+      std::ptr::null_mut(),
+      OPEN_EXISTING,
+      FILE_ATTRIBUTE_DIRECTORY,
+      std::ptr::null_mut(),
+    );
+    if file_handle == INVALID_HANDLE_VALUE {
+      eprintln!("SSDFSDFSDF");
```
Comment (aapoalas) [inline ext/fs/lib.rs:1351]: I see :D

File: ext/fs/30_fs.js (hunk 1 above)
Comment (bartlomieju) [reply ext/fs/30_fs.js:217]: No idea, @littledivy thoughts?

**Review by aapoalas — DISMISSED** [review]
lstating a dangling file seems to be failing on Windows. Presumably that's still related to these changes?

**Review by dsherret — APPROVED** [review]
LGTM!

**Comment by lucacasonato** [comment]
Yay! Can we revert the part of this change that sets everything to 0 for windows? The null->zero should happen in std/node. Also this is a breaking API change.

**Pull request merged**